external-builds/pytorch/run_linux_pytorch_tests.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+set -uxo pipefail
+
+# Validate environment variable
+if [ -z "$PYTHON_VERSION" ]; then
+  exit 1
+fi
+
+# Determine directories
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(realpath "$SCRIPT_DIR/../..")"
+PYTORCH_DIR="$ROOT_DIR/external-builds/pytorch/pytorch"
+
+# Activate virtual environment
+VENV_DIR="$ROOT_DIR/venv"
+if [ ! -d "$VENV_DIR" ]; then
+  exit 1
+fi
+source "$VENV_DIR/bin/activate" || exit 1
+
+# Set trap to deactivate virtual environment
+trap "deactivate 2> /dev/null" EXIT
+
+# Checkout repositories
+cd "$ROOT_DIR"
+./external-builds/pytorch/pytorch_torch_repo.py checkout --no-hipify --no-patch
+
+# Set environment variables
+export PYTORCH_PRINT_REPRO_ON_FAILURE=0
+export PYTORCH_TEST_WITH_ROCM=1
+export MIOPEN_CUSTOM_CACHE_DIR=/tmp/miopen-cache
+export PYTORCH_TESTING_DEVICE_ONLY_FOR="cuda"
+export PYTHONPATH="$PYTORCH_DIR/test:${PYTHONPATH:-}"
+
+# Run pytest
+cd "$ROOT_DIR"
+set +e
+python -m pytest \
+  "$PYTORCH_DIR/test/test_nn.py" \
+  "$PYTORCH_DIR/test/test_torch.py" \
+  "$PYTORCH_DIR/test/test_cuda.py" \
+  "$PYTORCH_DIR/test/test_ops.py" \
+  "$PYTORCH_DIR/test/test_unary_ufuncs.py" \
+  "$PYTORCH_DIR/test/test_binary_ufuncs.py" \
+  "$PYTORCH_DIR/test/test_autograd.py" \
+  "$PYTORCH_DIR/test/inductor/test_torchinductor.py" \
+  -v \
+  --continue-on-collection-errors \
```

Comment:
Are there collection errors?

Reply:
This was for debugging purposes -  will remove.